typing/jkind.ml

```diff
@@ -1574,19 +1646,24 @@ let of_new_legacy_sort_var ~why =
 
 let of_new_legacy_sort ~why = fst (of_new_legacy_sort_var ~why)
 
-let of_const ~annotation ~why (c : 'd Const.t) =
+let of_const (type l r) ~annotation ~why ~(quality : (l * r) jkind_quality)
+    (c : (l * r) Const.t) =
   { jkind = Layout_and_axes.map Layout.of_const c;
     annotation;
     history = Creation why;
-    has_warned = false
+    has_warned = false;
+    quality
   }
 
 let of_builtin ~why Const.Builtin.{ jkind; name } =
-  of_const ~annotation:(mk_annot name) ~why jkind |> allow_left |> allow_right
+  jkind |> Layout_and_axes.allow_left |> Layout_and_axes.disallow_right
+  |> of_const ~annotation:(mk_annot name) ~why ~quality:Best
 
 let of_annotated_const ~context ~annotation ~const ~const_loc =
   let context = Context_with_transl.get_context context in
-  of_const ~annotation ~why:(Annotated (context, const_loc)) const
+  of_const ~annotation
+    ~why:(Annotated (context, const_loc))
+    const ~quality:Not_best
```

Comment:
This choice of `Not_best` should be reflected in the mli, at least for `of_type_decl` and `of_type_decl_default`.